pkgs/by-name/ri/ringboard/package.nix

```diff
@@ -0,0 +1,91 @@
+{
+  rustPlatform,
+  fetchFromGitHub,
+  lib,
+  libxkbcommon,
+  libGL,
+  wayland,
+  xorg,
+  makeWrapper,
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "ringboard";
+
+  # release version needs nightly, so we use a custom tree, see:
+  # https://github.com/SUPERCILEX/clipboard-history/issues/22#issuecomment-3322075172
+  version = "0.12.2-unstable-2025-09-23";
+
+  src = fetchFromGitHub {
+    owner = "SUPERCILEX";
+    repo = "clipboard-history";
+    rev = "228a39dd8a9aece0bb06f68ad44906b297270628";
+    hash = "sha256-qA7wwvWnnZHN9edkmubEo37F+peU0LQGo/Zl8FpywuE=";
+  };
+
+  cargoHash = "sha256-MFfuUu/hpb6Uaqe21bvXNKRyJazAL5m+Vw/vAeeDYEk=";
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  buildInputs = [
+    libxkbcommon
+    libGL
+
+    # WINIT_UNIX_BACKEND=wayland
+    wayland
+
+    # WINIT_UNIX_BACKEND=x11
+    xorg.libXcursor
+    xorg.libXrandr
+    xorg.libXi
+    xorg.libX11
+  ];
+
+  buildPhase = ''
+     runHook preBuild
+
+    local flagsArray=("-j $NIX_BUILD_CORES --target x86_64-unknown-linux-gnu --offline --release");
```

Comment:
This should probably be done something with, so that it can compile on other platforms as well.

Reply:
I think there might also be `stdenv.hostPlatform.rust.rustcTarget` or something similar